src/AutoImplement/Program.cs

```diff
@@ -156,8 +156,8 @@ private static void GenerateImplementation<TPatternBuilder>(Type interfaceType)
                foreach (var member in FindAllMembers(interfaceType)) {
                   var metadata = new MemberMetadata(member);
                   switch (member.MemberType) {

```

Comment:
Why not use a list (or array) of supported MemberTypes, something like:
```suggestion
                  var supportedMemberTypes = new List<MemberTypes>(){MemberTypes.Method, MemberTypes.Event, MemberTypes.Property};
                  if(!supportedMemberTypes.Contains(member.MemberType)) {
                      throw new NotImplementedException($"AutoImplement has no way to implement a {member.MemberType} member.");
                  }
```

Reply:
Is the goal to completely remove the switch statement? If so, I'd need to have a dictionary, not a list, with `MemberTypes` as the key and `Action<MemberInfo, MemberMetadata, IPatternBuilder>` as the value, so that I could switch which method I call based on `member.MemberType`.

Right now, that doesn't seem worth it, since I'm only switching on `member.MemberType` once. But I'm interested in hearing your case.

Reply:
We spoke about this: looks like you misread the code. We decided that shouldn't change this.